internal/internal_workflow_client.go

```diff
@@ -343,6 +343,7 @@ func (wc *workflowClient) SignalWithStartWorkflow(ctx context.Context, workflowI
 		RetryPolicy:                         convertRetryPolicy(options.RetryPolicy),
 		CronSchedule:                        common.StringPtr(options.CronSchedule),
 		Memo:                                memo,
+		WorkflowIdReusePolicy:               options.WorkflowIDReusePolicy.toThriftPtr(),
```

Comment:
what do you think about adding unit tests to assert that we populate requests with correct values from higher level apis (by mocking service client) ?

Reply:
Good point.  Let me land this first and then I will look into adding a unit test in a separate PR.